mod-source-record-manager-server/src/main/java/org/folio/dao/JournalRecordDaoImpl.java

```diff
@@ -161,4 +174,33 @@ private String prepareSortingClause(String sortBy, String order) {
     return format(ORDER_BY_PATTERN, sortBy, order);
   }
 
+  private JobLogEntryDtoCollection mapRowSetToJobLogDtoCollection(RowSet<Row> rowSet) {
+    JobLogEntryDtoCollection jobLogEntryDtoCollection = new JobLogEntryDtoCollection()
+      .withTotalRecords(0);
+
+    rowSet.forEach(row -> {
+      JobLogEntryDto jobLogEntryDto = new JobLogEntryDto()
+        .withJobExecutionId(row.getValue("job_execution_id").toString())
+        .withSourceRecordId(row.getValue("source_id").toString())
+        .withSourceRecordOrder(row.getInteger("source_record_order"))
+        .withSourceRecordTitle(row.getString("title"))
+        .withSourceRecordActionStatus(mapNameToEntityActionStatus(row.getString("source_record_action_status")))
+        .withInstanceActionStatus(mapNameToEntityActionStatus(row.getString("instance_action_status")))
+        .withHoldingsActionStatus(mapNameToEntityActionStatus(row.getString("holdings_action_status")))
+        .withItemActionStatus(mapNameToEntityActionStatus(row.getString("item_action_status")))
+        .withOrderActionStatus(mapNameToEntityActionStatus(row.getString("order_action_status")))
+        .withInvoiceActionStatus(mapNameToEntityActionStatus(row.getString("invoice_action_status")))
+        .withError(row.getString("error"));
+
+      jobLogEntryDtoCollection
+        .withTotalRecords(row.getInteger("total_count"))
+        .getEntries().add(jobLogEntryDto);
+    });
+    return jobLogEntryDtoCollection;
+  }
+
+  private org.folio.rest.jaxrs.model.ActionStatus mapNameToEntityActionStatus(String name) {
+    return name == null ? null : org.folio.rest.jaxrs.model.ActionStatus.fromValue(name);
```

Comment:
@RuslanLavrov can there be the case when "name" will be NOT null, but EMPTY? If yes, then it would be better to use isBlank()-method.

Reply:
no

Reply:
@RuslanLavrov  Ok, thank you!